test/common/config/new_grpc_mux_impl_test.cc

```diff
@@ -88,28 +101,73 @@ class NewGrpcMuxImplTestBase : public testing::Test {
     EXPECT_CALL(async_stream_, sendMessageRaw_(Grpc::ProtoBufferEq(expected_request), false));
   }
 
+  void remoteClose() {
+    if (isUnifiedMuxTest()) {
+      dynamic_cast<XdsMux::GrpcMuxDelta*>(grpc_mux_.get())
```

Comment:
It may be better to only cast in the if clause, and then apply the `->grpcStreamForTest().onRemoteClose(Grpc::Status::WellKnownGrpcStatus::Canceled, "");` outside of the if clause. (Should improve maintenance down the road).
Also applies to the functions below.

Reply:
Not sure what you mean here: `grpcStreamForTest()` isn't in the mux interface, concrete mux types are needed to invoke it. There isn't a common base between `XdsMux::GrpcMuxDelta` and `NewGrpcMuxImpl`, I can't declare a local var to hold the result of such cast...